.github/workflows/ci.yaml

```diff
@@ -46,6 +53,9 @@ jobs:
         arguments: build
 
   gha-oidc-tests:
+    # Pull requests from different repositories have no access to `id-token: write`
```

Comment:
It will run on pushes to `master` branch anyway.

I think behind the lines of
1) adding `id-token: write` to the default tests
2) adding an extra option (e.g. env var) passed to tests like `OIDC_TESTS=ALWAYS_SKIP | SKIP_ON_MISSING_OIDC_TOKEN | FAIL_IF_TEST_FAILS` (naming is TBD). Then the same tests will verify OIDC when building master branch, and it would just skip when building in PRs without OIDC.

I've no solution yet, however, I would probably remove "many test tasks" in favour of a single one.
Yet another alternative is to move OIDC-related tests to a new project which would be skipped or ignored depending on OIDC presence.